gtars-bbcache/src/client.rs

```diff
@@ -10,42 +15,156 @@ use std::path::{Path, PathBuf};
+        // handle the cache dir
+        let raw_path_to_cache_folder = self.cache_folder.unwrap_or_else(get_default_cache_folder);
+        let raw_str_to_cache_folder = raw_path_to_cache_folder.to_string_lossy().into_owned();
+        let expanded_str = shellexpand::env(&raw_str_to_cache_folder)
+            .unwrap_or_else(|_| raw_str_to_cache_folder.clone().into())
+            .into_owned();
+        let abs_path_to_cache_folder = PathBuf::from(expanded_str);
+        create_dir_all(&abs_path_to_cache_folder)?;
+
+        // handle the bedbase api
+        let bedbase_api = self.bedbase_api.unwrap_or_else(get_default_bedbase_api);
+
+        // create sub folders
+        let bedfile_subfolder = &abs_path_to_cache_folder.join(DEFAULT_BEDFILE_SUBFOLDER);
         create_dir_all(bedfile_subfolder)?;
         let bedfile_cache = BioCache::new(bedfile_subfolder);
 
-        let bedset_subfolder = &cache_folder.join(DEFAULT_BEDSET_SUBFOLDER);
+        let bedset_subfolder = &abs_path_to_cache_folder.join(DEFAULT_BEDSET_SUBFOLDER);
         create_dir_all(bedset_subfolder)?;
         let bedset_cache = BioCache::new(bedset_subfolder);
 
         Ok(BBClient {
-            cache_folder,
+            cache_folder: abs_path_to_cache_folder,
             bedbase_api,
             bedfile_cache,
             bedset_cache,
         })
     }
+}
+
+/// Client for managing BED file and BED set caching from BEDbase.
+///
+/// `BBClient` provides a high-level interface for:
+/// - Downloading and caching BED files from the BEDbase API
+/// - Managing local BED file collections
+/// - Organizing BED sets (collections of related BED files)
+/// - Querying and removing cached resources
+///
+/// The client maintains two separate caches:
+/// - **bedfile_cache**: Individual BED files (stored as `.bed.gz`)
+/// - **bedset_cache**: BED set metadata files (stored as `.txt` with lists of BED IDs)
+///
+///
+/// # Examples
+///
+/// ```rust,no_run
+/// use gtars_bbcache::client::BBClient;
+///
+/// # fn main() -> anyhow::Result<()> {
+/// // Create a client with default settings
+/// let mut client = BBClient::builder().finish()?;
+///
+/// // Download and cache a BED file from BEDbase
+/// let region_set = client.load_bed("6b2e163a1d4319d99bd465c6c78a9741")?;
+///
+/// // Check if file exists in cache
+/// let path = client.seek("6b2e163a1d4319d99bd465c6c78a9741")?;
+/// println!("Cached at: {:?}", path);
+///
+/// // List all cached BED files
+/// let beds = client.list_beds()?;
+/// println!("Found {} cached BED files", beds.len());
+///
+/// // Remove from cache
+/// client.remove("6b2e163a1d4319d99bd465c6c78a9741")?;
+/// # Ok(())
+/// # }
+/// ```
+pub struct BBClient {
+    /// Path to the root cache directory
+    pub cache_folder: PathBuf,
+    /// BEDbase API endpoint URL
+    pub bedbase_api: String,
+    /// Internal cache manager for BED files
+    bedfile_cache: BioCache,
+    /// Internal cache manager for BED sets
+    bedset_cache: BioCache,
```

Comment:
Feel like some of these are just not necessary comments... Sort of visually messes with things